test/gtest/uct/ib/test_sockaddr.cc

```diff
@@ -855,6 +839,214 @@ UCS_TEST_P(test_uct_cm_sockaddr, connect_client_to_server_reject_with_delay)
+
+        if (m_all_eps[index].state & TEST_EP_FLAG_DISCONNECT_INITIATOR) {
+            pthread_mutex_lock(&m_lock);
+            m_ep_init_disconnect_cnt--;
+            pthread_mutex_unlock(&m_lock);
+        } else {
+            ASSERT_UCS_OK(uct_ep_disconnect(ep, 0));
+        }
+    }
+
+    static void server_disconnect_cb(uct_ep_h ep, void *arg) {
+        test_uct_cm_sockaddr_stress *self =
+                        reinterpret_cast<test_uct_cm_sockaddr_stress *>(arg);
+
+        self->common_test_disconnect(ep);
+        self->m_server_disconnect_cnt++;
+    }
+
+    static void client_disconnect_cb(uct_ep_h ep, void *arg) {
+        test_uct_cm_sockaddr_stress *self =
+                        reinterpret_cast<test_uct_cm_sockaddr_stress *>(arg);
+
+        self->common_test_disconnect(ep);
+        self->m_client_disconnect_cnt++;
+    }
+
+    void server_accept(entity *server, uct_conn_request_h conn_request,
+                       uct_ep_server_connect_cb_t connect_cb,
+                       uct_ep_disconnect_cb_t disconnect_cb,
+                       void *user_data)
+    {
+        server->accept(server->cm(), conn_request, connect_cb, disconnect_cb,
+                       user_data);
+    }
+
+    static void
+    conn_request_cb(uct_listener_h listener, void *arg,
+                    const char *local_dev_name,
+                    uct_conn_request_h conn_request,
+                    const uct_cm_remote_data_t *remote_data) {
+
+        test_uct_cm_sockaddr_stress *self =
+                        reinterpret_cast<test_uct_cm_sockaddr_stress *>(arg);
+
+        if (test_uct_cm_sockaddr::common_conn_request(listener, arg,
+                                                      local_dev_name,
+                                                      conn_request,
+                                                      remote_data)) {
+            self->server_accept(self->m_server, conn_request,
+                                server_connect_cb, server_disconnect_cb, self);
+        }
+    }
+
+protected:
+    int                     m_clients_num;
+    std::vector<ep_state_t> m_all_eps;
+    int                     m_ep_init_disconnect_cnt;
+    pthread_mutex_t         m_lock;
+};
+
+UCS_TEST_P(test_uct_cm_sockaddr_stress, many_clients_to_one_server)
+{
+    int i, no_disconnect_eps_cnt = 0;
+    entity *client_test;
+    time_t seed = time(0);
+    ucs_time_t deadline;
+
+    /* Listen */
+    start_listen(test_uct_cm_sockaddr_stress::conn_request_cb);
+
+    /* Connect */
+    /* multiple clients, each on a cm of its own, connecting to the same server */
+    for (i = 0; i < m_clients_num; ++i) {
+        client_test = uct_test::create_entity();
+        m_entities.push_back(client_test);
+
+        client_test->max_conn_priv = client_test->cm_attr().max_conn_priv;
+        client_test->connect(0, *m_server, 0, m_connect_addr, client_priv_data_cb,
+                             client_connect_cb, client_disconnect_cb, this);
```

Comment:
we are passing client_disconnect_cb here but initializing the fields which this function accesses only at line 972. is this ok?

Reply:
@yosefe why not? 
this is the disconnect_cb so it won't be invoked before we start the disconnect flow - which happens after line 972

Reply:
since this is a test, i'd suggest to at least have some check disconnect is not called too early, in case there is bug in uct

Reply:
if disconnect is called early then the test wouldn't complete and fail. all the checks below will fail

Reply:
ok